src/bundle/Controller/UrlAliasController.php

```diff
@@ -66,24 +68,60 @@ public function addAction(Request $request): Response
                     $data->isRedirect()
                 );
 
-                return new RedirectResponse($this->generateUrl('_ezpublishLocation', [
-                    'locationId' => $data->getLocation()->id,
-                    '_fragment' => UrlsTab::URI_FRAGMENT,
-                ]));
+                return $this->redirectToLocation($data->getLocation());
             });
 
             if ($result instanceof Response) {
                 return $result;
             }
         }
 
-        $redirectionUrl = null !== $location
-            ? $this->generateUrl('_ezpublishLocation', [
+        return $this->redirectToLocation($location);
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function removeAction(Request $request): Response
+    {
+        $form = $this->formFactory->removeCustomUrl();
+        $form->handleRequest($request);
+
+        $location = $form->getData()->getLocation();
+
+        if ($form->isSubmitted()) {
+            $result = $this->submitHandler->handle($form, function (CustomUrlRemoveData $data) {
+                $aliasToRemoveList = [];
+                foreach ($data->getUrlAliases() as $customUrlId => $selected) {
+                    $aliasToRemoveList[] = $this->urlAliasService->load($customUrlId);
+                }
+                $this->urlAliasService->removeAliases($aliasToRemoveList);
+
+                return $this->redirectToLocation($data->getLocation());
+            });
+
+            if ($result instanceof Response) {
+                return $result;
+            }
+        }
+
+        return $this->redirectToLocation($location);
+    }
+
+    /**
+     * @param Location|null $location
+     *
+     * @return RedirectResponse
+     */
+    private function redirectToLocation(?Location $location): RedirectResponse
```

Comment:
This method is not clear to me. It's named "redirectToLocation" but in fact it can redirect to dashboard. It also allows to pass `null` $location. `redirectToLocation` is a great addition but it should be a part of base `Controller` so we can reuse it.